spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -246,75 +264,91 @@
     end
 
     it "donates if there are sub-programs running" do
-      Strand.create(parent_id: st.id, prog: "BootstrapRhizome", label: "start", stack: [{}], lease: Time.now + 10)
+      Strand.create(parent_id: server_strand.id, prog: "BootstrapRhizome", label: "start", stack: [{}], lease: Time.now + 10)
       expect { nx.wait_bootstrap_rhizome }.to nap(5)
     end
   end
 
   describe "#mount_data_disk" do
-    it "formats data disk if format command is not sent yet or failed" do
-      expect(postgres_server).to receive(:storage_device_paths).and_return(["/dev/vdb"])
-      expect(sshable).to receive(:d_run).with("format_disk", "sudo", "mkfs", "--type", "ext4", "/dev/vdb")
+    # Create storage volumes for the VM (not created automatically by Vm::Nexus.assemble in test mode)
+    let(:data_volume) {
+      VmStorageVolume.create(vm_id: postgres_server.vm.id, boot: false, size_gib: 64, disk_index: 1)
+    }
 
+    it "formats data disk if format command is not sent yet or failed" do
+      data_volume
+      device_path = postgres_server.storage_device_paths.first
       expect(sshable).to receive(:d_check).with("format_disk").and_return("NotStarted")
+      expect(sshable).to receive(:d_run).with("format_disk", "sudo", "mkfs", "--type", "ext4", device_path)
       expect { nx.mount_data_disk }.to nap(5)
     end
 
     it "formats data disk correctly when there are multiple storage volumes" do
-      expect(postgres_server).to receive(:storage_device_paths).and_return(["/dev/nvme1n1", "/dev/nvme2n1"])
-      expect(sshable).to receive(:_cmd).with("sudo mdadm --create --verbose /dev/md0 --level=0 --raid-devices=2 /dev/nvme1n1 /dev/nvme2n1")
-      expect(sshable).to receive(:d_run).with("format_disk", "sudo", "mkfs", "--type", "ext4", "/dev/md0")
+      # RAID setup only happens on AWS where storage_device_paths returns multiple paths.
+      # storage_device_paths does SSH to lsblk on AWS - stubbing to avoid AWS infra setup.
+      paths = ["/dev/nvme1n1", "/dev/nvme2n1"]
+      allow(postgres_server).to receive(:storage_device_paths).and_return(paths)
 
       expect(sshable).to receive(:d_check).with("format_disk").and_return("NotStarted")
+      expect(sshable).to receive(:cmd).with("sudo mdadm --create --verbose /dev/md0 --level=0 --raid-devices=:count :shelljoin_storage_device_paths",
```

Comment:
```suggestion
      expect(sshable).to receive(:_cmd).with("sudo mdadm --create --verbose /dev/md0 --level=0 --raid-devices=:count :shelljoin_storage_device_paths",
```

Same for any other cases where `cmd` is currently mocked.